### PR TITLE
feat: migrate list responses to new enum pattern

### PR DESCRIPTION
**Pull request by anitarua**
Closes https://github.com/momentohq/dev-eco-issue-tracker/issues/846

### CODE REVIEW

File: packages/common-integration-tests/src/list.ts [hunk 1]
```diff
@@ -619,7 +621,7 @@ export function runListTests(
 
         expect(popResponse.value()).toEqual('foo');
 
-        const hitResponse = popResponse as CacheListPopBack.Hit;
+        const hitResponse = popResponse as CacheListPopFront.Hit;
```
Comment (bruuuuuuuce) [inline packages/common-integration-tests/src/list.ts:624]: another one  💪

File: packages/core/src/messages/responses/cache-list-pop-front.ts [hunk 2]
```diff
@@ -1,62 +1,45 @@
...
-export abstract class Response extends ResponseBase {
-  public value(): string | undefined {
-    if (this instanceof Hit) {
-      return (this as Hit).value();
-    }
-    return undefined;
-  }
+interface IResponse {
+  value(): string | undefined;
+  type: CacheListPopFrontResponse;
```
Comment (bruuuuuuuce) [inline packages/core/src/messages/responses/cache-list-pop-front.ts:14]: I just learned that you can make this value on the interface `readonly`.

```
interface IResponse {
  value(): string | undefined;
  readonly type: CacheListPopFrontResponse;
}
```

 I think we should do that, so we don't accidentally make it mutable inside of the class. I can work on changing the existing ones to be readonly.

Comment (anitarua) [reply packages/core/src/messages/responses/cache-list-pop-front.ts:14]: oh nice! yeah I can update the ones here to be readonly

**Review by bruuuuuuuce — APPROVED** [review]
:shipit:

**Pull request merged**